clevertap.js

```diff
@@ -1948,7 +1948,7 @@
             continue;
           }
 
-          if (profileKey === 'Gender' && !profileVal.match(/^M$|^F$/)) {
+          if (profileKey === 'Gender' && !profileVal.match(/\b(?:[mM](?:ale)?|[fF](?:emale)?|[oO](?:thers)?|[uU](?:nknown)?)\b/)) {
             valid = false;
```

Comment:
Would it be better if we make the check more readable instead of using regex here? Just a suggestion.

Something like 


```js
const validGenders = ['male', 'female', 'others', 'unknown', 'm', 'f']; // can be in a constants 
const isMatch = validGenders.includes(profileVal.toLowerCase());
```